src/main/resources/static/js/ltistorage.js

```diff
@@ -0,0 +1,517 @@
+          log.error({
+            message:
+              "Ignoring message, invalid subject: [" +
+              event.data.subject +
+              "] expected: [" +
+              data.subject +
+              ".response]",
+          });
+          return log.print();
+        }
+        window.removeEventListener("message", messageHandler);
+        clearTimeout(timeout);
+        if (event.data.error) {
+          log.error(event.data.error);
+          log.print();
+          return reject(event.data.error);
+        }
+        log.print();
+        resolve(event.data);
+      };
+      window.addEventListener("message", messageHandler);
+      log.request(targetFrameName, data, targetOrigin);
+      targetFrame.postMessage(data, targetOrigin);
+      timeout = setTimeout(() => {
+        window.removeEventListener("message", messageHandler);
+        let timeout_error = {
+          code: "timeout",
+          message: "No response received after 1000ms",
+        };
+        log.error(timeout_error);
+        log.print();
+        reject(timeout_error);
+      }, 1000);
+    });
+  }
+  async sendPostMessageIfCapable(data) {
+    // Call capability service
+    return Promise.any([
+      this.sendPostMessage(
+        { subject: "lti.capabilities" },
+        __classPrivateFieldGet(
+          this,
+          _LtiPostMessage_instances,
+          "m",
+          _LtiPostMessage_getTargetWindow
+        ).call(this),
+        "*"
+      ),
+      // Send new and old capabilities messages for support with pre-release subjects
+      this.sendPostMessage(
+        { subject: "org.imsglobal.lti.capabilities" },
+        __classPrivateFieldGet(
+          this,
+          _LtiPostMessage_instances,
+          "m",
+          _LtiPostMessage_getTargetWindow
+        ).call(this),
+        "*"
+      ),
+    ]).then((capabilities) => {
+      if (typeof capabilities.supported_messages == "undefined") {
+        return Promise.reject({
+          code: "not_found",
+          message: "No capabilities",
+        });
+      }
+      for (let i = 0; i < capabilities.supported_messages.length; i++) {
+        if (
+          ![data.subject, "org.imsglobal." + data.subject].includes(
+            capabilities.supported_messages[i].subject
+          )
+        ) {
+          continue;
+        }
+        // Use subject specified in capabilities for backwards compatibility
+        data.subject = capabilities.supported_messages[i].subject;
+        return this.sendPostMessage(
+          data,
+          __classPrivateFieldGet(
+            this,
+            _LtiPostMessage_instances,
+            "m",
+            _LtiPostMessage_getTargetWindow
+          ).call(this),
+          undefined,
+          capabilities.supported_messages[i].frame
+        );
+      }
+      return Promise.reject({
+        code: "not_found",
+        message: "Capabilities not found",
+      });
+    });
+  }
+  async putData(key, value) {
```

Comment:
I modified this function because I didn't understand what the original function from 1EdTech was meant to do, so now it is returning the Object containing the data for the postMessage.